http_test.go

```diff
@@ -156,3 +161,27 @@ func TestRedirectNotWhenHTTPS(t *testing.T) {
 
 	assert.Equal(t, http.StatusOK, res.StatusCode, "status code should be %d, got: %d", http.StatusOK, res.StatusCode)
 }
+
+func TestGracefulShutdown(t *testing.T) {
+	signals := []syscall.Signal{syscall.SIGINT, syscall.SIGTERM}
+
+	for i, signal := range signals {
+		name := fmt.Sprintf("%s", signal)
+		t.Run(name, func(t *testing.T) {
+			opts := NewOptions()
+			opts.HTTPAddress = fmt.Sprintf(":%d", 4180+i)
+			srv := Server{Handler: http.DefaultServeMux, Opts: opts}
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				srv.ServeHTTP()
+			}()
+			time.Sleep(500 * time.Millisecond)
+			if err := syscall.Kill(os.Getpid(), signal); err != nil {
```

Comment:
Is this sending a kill signal to the test suite? Does this not cause the test suite to fail? Is there no other way to simulate the signal?